x-pack/plugins/security/public/management/roles/edit_role/spaces_popover_list/spaces_popover_list.tsx

```diff
@@ -75,133 +82,84 @@ export class SpacesPopoverList extends Component<Props, State> {
       }),
     };
 
-    if (this.props.spaces.length >= SPACE_SEARCH_COUNT_THRESHOLD) {
-      return (
-        <EuiContextMenuPanel {...panelProps}>
-          {this.renderSearchField()}
-          {this.renderSpacesListPanel(items, searchTerm)}
-        </EuiContextMenuPanel>
-      );
-    }
-
-    return <EuiContextMenuPanel {...panelProps} items={items} />;
+    return (
+      <EuiSelectable
+        {...panelProps}
+        searchable={this.props.spaces.length >= SPACE_SEARCH_COUNT_THRESHOLD}
+        searchProps={
+          this.props.spaces.length >= SPACE_SEARCH_COUNT_THRESHOLD
+            ? ({
+                placeholder: i18n.translate(
+                  'xpack.security.management.editRole.spacesPopoverList.findSpacePlaceholder',
+                  {
+                    defaultMessage: 'Find a space',
+                  }
+                ),
+                compressed: true,
+                isClearable: true,
+                id: 'spacesPopoverListSearch',
+              } as any)
```

Comment:
We're trying to avoid `any` since it means we don't get type safety but I don't have a better fix to the typings issue in `EuiSelectable` here so think this is acceptable.
 
I would reach out to the team or create an issue but we don't need to block this from merging.

Reply:
Larry and I talked about this as well. I will ping EUI for a recommendation, and create an issue so we don't lose track of it.

Reply:
I removed the use of 'any' and used an ExclusiveUnion keyed by the searchable condition (SPACE_SEARCH_COUNT_THRESHOLD) to create the searchable and search props.
In the future we could utilize the EuiSelectableSearchableProps type, but it is not exported at this time. There is a PR to export this type [here](https://github.com/elastic/eui/pull/6096).